x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchTask.java

```diff
@@ -401,16 +400,15 @@ public void onPartialReduce(List<SearchShard> shards, TotalHits totalHits,
                 reducedAggs = () -> null;
             } else {
                 /*
-                 * Keep a reference to the serialized form of the partially
-                 * reduced aggs and reduce it on the fly when someone asks
+                 * Keep a reference to the partially reduced aggs and reduce it on the fly when someone asks
                  * for it. It's important that we wait until someone needs
                  * the result so we don't perform the final reduce only to
                  * throw it away. And it is important that we keep the reference
-                 * to the serialized aggregations because SearchPhaseController
+                 * to the aggregations because SearchPhaseController
                  * *already* has that reference so we're not creating more garbage.
                  */
                 reducedAggs = () ->
-                    InternalAggregations.topLevelReduce(singletonList(aggregations.expand()), aggReduceContextSupplier.get());
```

Comment:
what is the rationale behind the choice of keeping the de-serialized partial aggs around rather then serializing them and expanding them later when needed?

Reply:
Because serializing and de-serializing is slow and makes the memory consumption worst when expanding. We can maybe revise this later but I feel like the real difference would be to serialize on a temporary file or in an index in case of async_search.

Reply:
I see, so this particular change does not really have to do with circuit breaking, but more around the feeling that keeping aggs around as they are is better overall? Was this measured out of curiosity? I think I am asking because I wonder why we were serializing/deserializing before

Reply:
> Was this measured out of curiosity? 

Yes, see https://github.com/elastic/elasticsearch/pull/62223#issuecomment-692928324

Reply:
++ great thanks

Reply:
one thing that could have made things clearer would be to have two separate PRs for the two changes, but it's fine as-is at this point, and I am maybe missing some reason why they were made as part of a single PR.